docs: undeprecate plugin (#70)
apache/cordova-plugin-device-motion: README.md

# cordova-plugin-device-motion

# Usage Notice

With the [W3C Device Orientation API](https://www.w3.org/TR/orientation-event/), Android, iOS, and Windows devices may not need this plugin anymore.

> However, on iOS 13+ potential issues with permissions and secure contexts can arise. Therefore, it is recommended to use this plugin as it uses a native implementation.

Reviewer: ```suggestion
However, on iOS 13+, potential issues with permissions and secure contexts can arise. Therefore it is recommended to use this plugin as it uses a native implementation.
```

# cordova-plugin-device-motion

# Usage Notice

With the [W3C Device Orientation API](https://www.w3.org/TR/orientation-event/), Android, iOS, and Windows devices may not need this plugin anymore.

However, for iOS 13+, a [secure context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) is required. Apple does not allow access to the device’s motion or the permission request from an insecure browsing context.

> If you use a custom scheme, i.e., the `app` protocol, for your iOS app, the browsing context is considered insecure. To solve this problem and be able to use the device motion feature, this plugin will be necessary on iOS 13+.

Reviewer 2: This is not true, on iOS 13 it's required to request a permission with `DeviceMotionEvent.requestPermission();`, but works just fine after doing it, doesn't matter if you use `file://` or `app://`.
Nevertheless, the permission request prompt has a few issues, see the mail I sent on the discuss thread for more information.

Author: Are you sure? My test app is running on `ionic://my-domain.com` and on user interaction the request was denied with `Call to requestPermission() failed, reason: Browsing context is not secure.`

Reviewer 2: then they might allow localhost, I tested on `app://localhost`

Reviewer: > Locally-delivered resources such as those with http://127.0.0.1 URLs, http://localhost URLs (under certain conditions), and file:// URLs are also considered to have been delivered securely.
> 
> Resources that are not local, to be considered secure, must meet the following criteria:
> 
> * must be served over https:// or wss:// URLs
> * the security properties of the network channel used to deliver the resource must not be considered deprecated

Maybe it is the `localhost` that is making the results different. If you ask Apple, I am sure they will claim that as a bug and will include the scheme as part of the check.

Reviewer 2: With Apple you can never be sure, geolocation doesn't work because WebView considers it not secure. So they might have forgot to add the scheme check for `DeviceMotionEvent.requestPermission();`, or they are switching to consider custom schemes secure. It still works on iOS 14 beta.

Author: ```suggestion
However, for iOS 13+, a [secure context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) is required. Apple does not allow access to the device’s motion or the permission request from an insecure browsing context. Locally-delivered resources such as those with http://127.0.0.1 URLs, http://localhost URLs (under certain conditions), and file:// URLs are also considered to have been delivered securely.

If you use a custom hostname, i.e., something else than `localhost`, for your iOS app, the browsing context is considered insecure. To solve this problem and be able to use the device motion feature, this plugin will be necessary on iOS 13+.
```

How about this? I'd like to avoid gambling's about Tim Apple's future decisions if they forgot to check the protocol or not.

Reviewer 2: I think we don't need to enter into all those details, we can just say on iOS 13+ it could present some problems and it's recommended to use the plugin as it uses a native implementation.

Author: ```suggestion
However, on iOS 13+ potential issues with permissions and secure contexts can arise. Therefore, it is recommended to use this plugin as it uses a native implementation.
```

This is much shorter now :)

Reviewer 2: looks good!
